dap2-service/src/main/scala/latis/service/dap2/Dds.scala

```diff
@@ -0,0 +1,100 @@
+import latis.util.LatisException
+
+final case class Dds(id: Identifier, typeDecls: List[Dds.TypeDecl]) {
+  override val toString: String = {
+    val prefix = Doc.text("Dataset {")
+    val suffix = Doc.char('}') + Doc.space + Doc.text(id.asString) + Doc.char(';')
+    val body = Doc.intercalate(Doc.line, typeDecls.map {
+      _.toDoc
+    }).indent(2)
+    val doc = prefix / body / suffix
+    doc.render(1)
+  }
+}
+
+object Dds {
+  sealed trait TypeDecl {
+    def toDoc: Doc
+  }
+  final case class AtomicDecl(id: Identifier, ty: AtomicType) extends TypeDecl {
+    override val toDoc: Doc = Doc.str(ty) + Doc.space + Doc.text(id.asString) + Doc.char(';')
+  }
+  final case class ArrayDecl(/* TODO: Not needed yet */)
+  final case class StructureDecl(id: Identifier, typeDecls: List[TypeDecl]) extends TypeDecl {
+    override val toDoc = {
+      val prefix = Doc.text("Structure {")
+      val suffix = Doc.char('}') + Doc.space + Doc.text(id.asString) + Doc.char(';')
+      val body = Doc.intercalate(Doc.line, typeDecls.map { _.toDoc }).indent(2)
+      prefix / body / suffix
+    }
+  }
+  final case class SequenceDecl(id: Identifier, typeDecls: List[TypeDecl]) extends TypeDecl {
+    override val toDoc = {
+      val prefix = Doc.text("Sequence {")
+      val suffix = Doc.char('}') + Doc.space + Doc.text(id.asString) + Doc.char(';')
+      val body = Doc.intercalate(Doc.line, typeDecls.map {
+        _.toDoc
+      }).indent(2)
+      prefix / body / suffix
+    }
+  }
+  final case class GridDecl(/* TODO: Not needed yet */)
+
+  sealed trait AtomicType {
+    override def toString = this.getClass.getSimpleName
+  }
+  final case class Byte() extends AtomicType
+  final case class Int16() extends AtomicType
+  final case class UInt16() extends AtomicType
+  final case class Int32() extends AtomicType
+  final case class UInt32() extends AtomicType
+  final case class Float32() extends AtomicType
+  final case class Float64() extends AtomicType
+  final case class String() extends AtomicType
+  final case class Url() extends AtomicType
+
+  def fromScalar(scalar: Scalar): Either[LatisException, AtomicDecl] = {
+    val id = scalar.id
+    scalar.valueType match {
+      case _: DoubleValueType.type => Right(AtomicDecl(id, Float64()))
+      case _: FloatValueType.type => Right(AtomicDecl(id, Float32()))
+      case _: IntValueType.type => Right(AtomicDecl(id, Int32()))
+      case _: ShortValueType.type => Right(AtomicDecl(id, Int16()))
+      case _: ByteValueType.type => Right(AtomicDecl(id, Byte()))
+      case _: StringValueType.type => Right(AtomicDecl(id, String()))
+      case _ => Left(LatisException("Scalar could not be parsed to a DDS atomic type."))
+    }
+  }
+
+  def fromTuple(tuple: Tuple): Either[LatisException, StructureDecl] = {
+    tuple.elements.map { elem => fromDataType(elem) }
+      .traverse(identity)
+      .map { lst => StructureDecl(tuple.id.getOrElse(id"unknown"), lst) }
+  }
```

Comment:
Even though one can (usually) do either, I typically prefer to preserve the curly brackets for cases when they're needed (multiple expressions separated by semicolons on a single line) or when something spans multiple lines. If it fits on one line, I would just use parens.

It's also common to omit the top level curly braces when the definition of a method is a single expression.

I also typically omit arguments from lambdas when the compiler can figure them out. I like to avoid naming things if I don't have to.

Also, `.traverse(identity) = .sequence`, and `.map(f).sequence = .traverse(f)`, so `.map(f).traverse(identity) = .traverse(f)`.

```suggestion
  def fromTuple(tuple: Tuple): Either[LatisException, StructureDecl] =
    tuple.elements.traverse(fromDataType).map {
      StructureDecl(tuple.id.getOrElse(id"unknown"), _)
    }
```